src/audio/volume/volume_ipc4.c

```diff
@@ -247,7 +245,10 @@ static int volume_set_volume(struct processing_module *mod, const uint8_t *data,
 			volume_set_chan(mod, cdata.channel_id,
 					cd->tvolume[cdata.channel_id], true);
 		}
-		if (cd->volume[cdata.channel_id] != cd->tvolume[cdata.channel_id])
+	}
+
+	for (i = 0; i < channels_count; i++) {
+		if (cd->volume[i] != cd->tvolume[i])
```

Comment:
"In IPC4 the individual channel gains are passed in separate messages."

Is this really something we want to support?

I don't really see how this might work if someone wants to e.g. implement a pan effect by moving a sound from the left to the right in a continuous motion?

Isn't it a classic requirement that such changes need to be handled in an 'atomic' way where all parameters are updated at the same time?

Maybe my perception is twisted by attending SDCA meetings where such volume/EQ changes are handled with a 'commit register' with all parameters flipped at once. Or I must be missing something...

Reply:
We could have a new "linux volume" component with different UUID to overcome this limitation while "gain" can be kept as-is an IPC4 compatible. The peakvolume/gain also has overhead from unused features in Linux.

There is already support for mute switch (that Windows is not using) that can control any number of channels. The same kind of ALSA mixercontrol (vs. gain/peakvolume set large config) could control the volume with no delays from channel to channel. But this would need a new kernel.